lib/src/util/coap_util.dart

```diff
@@ -7,88 +7,8 @@
 
 part of coap;
 
-/// Cancellable asynchronous sleep support class
-class CoapCancellableAsyncSleep {
-  CoapCancellableAsyncSleep(this._timeout);
-
-  final Duration _timeout;
-
-  /// Timeout
-  Duration get timeout => _timeout;
-
-  /// The completer
-  final Completer<void> _completer = Completer<void>();
-
-  /// The timer
-  late Timer _timer;
-
-  /// Timer running flag
-  bool _running = false;
-
-  /// Running
-  bool get isRunning => _running;
-
-  /// Start the timer
-  Future<void> sleep() {
-    if (!_running) {
-      _timer = Timer(_timeout, _timerCallback);
-      _running = true;
-    }
-    return _completer.future;
-  }
-
-  /// Cancel the timer
-  void cancel() {
-    if (_running) {
-      _timer.cancel();
-      _running = false;
-      _completer.complete();
-    }
-  }
-
-  /// The timer callback
-  void _timerCallback() {
-    _running = false;
-    _completer.complete();
-  }
-}
-
 /// Utility methods
 class CoapUtil {
```

Comment:
I think in this class, you could also remove the `regIP` member and use `InternetAddress.tryParse(host) == null` to perform the IP address check.

Reply:
The remaining `lookupHost` method could then be moved to the `CoapClient` class.